### PR TITLE
JeOS: expand image information

### PR DESCRIPTION
**Pull request by jlausuch**
Related tickets: 
- https://jira.suse.com/browse/TEAM-5808
- https://progress.opensuse.org/issues/110788

VR: https://openqa.opensuse.org/tests/2338516

### CODE REVIEW

**Comment by jlausuch** [comment]
```
select * from image_size
1652191977687102429 x86_64  20220509 JeOS-for-kvm-and-xen     openSUSE-Tumbleweed-Minimal-VM.x86_64-1.0.0-kvm-and-xen-Snapshot20220509.qcow2 https://openqa.opensuse.org/tests/2338516 uncompressed 216.125         Tumbleweed
```

File: tests/jeos/image_info.pm
Comment (Vogtinator) [inline tests/jeos/image_info.pm:115]: IMO not that useful, except maybe `/usr` and `/var`

File: tests/jeos/image_info.pm
Comment (Vogtinator) [inline tests/jeos/image_info.pm:136]: `It's always set, but if not ...` doesn't make much sense. Also: `if when no specified`

File: tests/jeos/image_info.pm
Comment (Vogtinator) [inline tests/jeos/image_info.pm:196]: Double empty line

File: schedule/jeos/sle/rpi/jeos-main.yaml
Comment (Vogtinator) [inline schedule/jeos/sle/rpi/jeos-main.yaml:6]: Question is how much the `firstrun` step causes the HDD to grow. Can you check whether that's significant?

Comment (jlausuch) [reply schedule/jeos/sle/rpi/jeos-main.yaml:6]: I'm getting the image size from the worker pool directory, e.g. https://openqa.opensuse.org/tests/2337465#step/image_info/2
 `-rw------- 2 _openqa-worker nogroup 226623488 May 10 07:36 openSUSE-Tumbleweed-Minimal-VM.x86_64-1.0.0-kvm-and-xen-Snapshot20220509.qcow2`
 
 But this file is used to boot the VM directly,
 `[2022-05-10T07:36:22.359786+02:00] [debug] running `/usr/bin/qemu-img create -f qcow2 -F qcow2 -b /var/lib/openqa/pool/11/openSUSE-Tumbleweed-Minimal-VM.x86_64-1.0.0-kvm-and-xen-Snapshot20220509.qcow2 /var/lib/openqa/pool/11/raid/hd0-overlay0 25769803776`
 
so it doesn't really matter, this qcow2 won't change.

File: tests/jeos/image_info.pm
Comment (jlausuch) [reply tests/jeos/image_info.pm:159]: That's up to decide ourselves, I can change it back to soft_fail, but I don't want that Database operations cause failures or soft-failures to the job, as they are not "product" related failures.

File: tests/jeos/image_info.pm
Comment (jlausuch) [reply tests/jeos/image_info.pm:136]: Even if you don't inject manually `CASEDIR`, but default all jobs have that variable by default.
See https://openqa.opensuse.org/tests/2337465/file/vars.json

File: tests/jeos/image_info.pm
Comment (jlausuch) [reply tests/jeos/image_info.pm:115]: Yeah, but as I'm running that in some loop way for all dirs, we can always chose to represent which data to display in the Graphs. Otherwise I would need to create a list of directories to analyze which may vary in the future. This way we just collect data for all directories under `/` and later one we show what we want.

File: tests/jeos/image_info.pm
Comment (jlausuch) [reply tests/jeos/image_info.pm:134]: Not bad idea, will try it out.

File: tests/jeos/image_info.pm
Comment (jlausuch) [reply tests/jeos/image_info.pm:133]: `my $dir = getcwd;` It's the current job pool directory on the worker, e.g. https://openqa.opensuse.org/tests/2337465#step/image_info/1
In this directory, the os-autoinst-distri-opensuse repo is copied, and all the data files are there, which is where  `data_url` is pointing to.

File: tests/jeos/image_info.pm
Comment (Vogtinator) [reply tests/jeos/image_info.pm:136]: In that case you can just use `get_required_var` like in other places and remove the comment

File: tests/jeos/image_info.pm
Comment (Vogtinator) [reply tests/jeos/image_info.pm:115]: You already exclude `/proc` and you'd have to exclude some others anyway as the sizes are nonsensical (`/sys`, `/dev/`, `/.snapshots`)...

File: tests/jeos/image_info.pm
Comment (Vogtinator) [reply tests/jeos/image_info.pm:133]: So you can just omit `$dir/` and the variable entirely?

File: tests/jeos/image_info.pm
Comment (Vogtinator) [reply tests/jeos/image_info.pm:159]: IMO it should even be a hard fail as the test couldn't do its job

File: schedule/jeos/sle/rpi/jeos-main.yaml
Comment (Vogtinator) [reply schedule/jeos/sle/rpi/jeos-main.yaml:6]: To clarify, you mean `isn't used to boot the VM directly`?

File: tests/jeos/image_info.pm
Comment (jlausuch) [reply tests/jeos/image_info.pm:159]: ok, will make it soft fail so we can react on it faster.

File: tests/jeos/image_info.pm
Comment (jlausuch) [reply tests/jeos/image_info.pm:133]: yes, I was testing this also on my local env and it always copies to that pool directory. Anyway, I'm gonna try `type_password`, so I might remove this.

File: schedule/jeos/sle/rpi/jeos-main.yaml
Comment (jlausuch) [reply schedule/jeos/sle/rpi/jeos-main.yaml:6]: yes, sorry. `it isn't used to boot` is the right sentence :)

File: tests/jeos/image_info.pm
Comment (jlausuch) [reply tests/jeos/image_info.pm:115]: ok, will get the info of those 2 dirs then, thanks for the feedback!

**Comment by jlausuch** [comment]
All suggestions have been addressed.
New VR: https://openqa.opensuse.org/tests/2344846

File: tests/jeos/image_info.pm
Comment (Vogtinator) [reply tests/jeos/image_info.pm]: Actually, I also forgot about usr-merge only being in TW... For SLE we'd need `/(s)bin` and `/lib(64)` as well. I also forgot about `/boot`... So excluding "special" mounts is probably the best idea.

Comment (jlausuch) [reply tests/jeos/image_info.pm]: ok, moving back to my original idea :D

**Comment by jlausuch** [comment]
@Vogtinator done: https://openqa.opensuse.org/tests/2345414#step/image_info/12

**Comment by jlausuch** [comment]
With latest changes: https://openqa.opensuse.org/tests/2345457#
Before merge, I need to remove this comment `return if (get_required_var('CASEDIR') !~ m/^sle$|^opensuse$/);`, so please don't merge before that.

File: tests/jeos/image_info.pm
Comment (mloviska) [inline tests/jeos/image_info.pm:90]: I think it would be safer to `ref $args{extra} eq 'HASH'`

**Review by mloviska — APPROVED** [review]
LGTM, just neat picking comments :)

File: tests/jeos/image_info.pm
Comment (jlausuch) [reply tests/jeos/image_info.pm:25]: Yes, looks like I can simply use `my`, as this `.pm` is not a package. I thought I needed global variables so that they can be accessed by all methods and be able to change them. Thanks!

File: tests/jeos/image_info.pm
Comment (jlausuch) [reply tests/jeos/image_info.pm:90]: It won't make a difference. This will `undef` if `extra` argument is not set.

```
my %my_hash1;
my %my_hash2;

$my_hash1{extra} = {rpm => 'myrpm'};

if ($my_hash1{extra}) {
    print("1\n");
}
if ($my_hash2{extra}) {
    print("2\n");
}
```
This will print only `1`

Comment (mloviska) [reply tests/jeos/image_info.pm:90]: From `undef` perspective, you are right. However, you wish to iterate over the keys of the hash, therefore you want to ensure you are dealing with a hash.

I also find it codewise descriptive rather than plain `$args{extra}`, which can actually hold any scalar or reference.

**Comment by ggardet** [comment]
This PR breaks tests on RPi 2/3/4 on O3: https://progress.opensuse.org/issues/111362

**Pull request merged**
